### PR TITLE
Add auth retries

### PR DESCRIPTION
**Pull request by giovannicuriel**

### CODE REVIEW

File: dojot/module/messenger.py [hunk 1]
```diff
@@ -94,22 +94,25 @@ def init(self):
...
-                LOGGER.info("%s bootstrapped.", ten)
-                LOGGER.debug("tenants: %s", self.tenants)
-            LOGGER.info("Finished tenant boostrapping")
-        except Exception as error:
-            LOGGER.warning("Could not get list of tenants: %s", error)
+        ret_tenants = auth.get_tenants()
+        if ret_tenants is None:
+            LOGGER.error("Cannot initialize messenger, as the list of tenants could not be retrieved.")
+            LOGGER.error("Bailing out.")
+            raise UserWarning("Cannot initialize messenger")
```
Comment (giovannicuriel) [inline dojot/module/messenger.py]: Didn't come up with a better exception type. Any suggestions?

Comment (matheuscampanhaf) [reply dojot/module/messenger.py]: Then, if ret_tenants is None, messenger will not initialize and the only way to try it again is restarting the container, right ? And isn't it better "Could not initialize messenger?"

File: dojot/module/auth.py [hunk 2]
```diff
@@ -45,11 +46,39 @@ def get_tenants(self):
...
+
         url = self.config.auth['url'] + "/admin/tenants"
-        ret = requests.get(url, headers={'authorization': "Bearer " + self.get_management_token()})
-        payload = ret.json()
+        retry_counter = self.config.auth["connection_retries"]
+        payload = None
+        while retry_counter > 0:
+            try:
+                LOGGER.debug("Retrieving list of tenants from auth...")
+                ret = requests.get(
```
Comment (raulnegreiros) [inline dojot/module/auth.py]: I'm not 100% about it, but you need call r.raise_for_status() for check error, isn't?

File: dojot/module/auth.py [hunk 3]
```diff
@@ -45,11 +46,39 @@ def get_tenants(self):
...
+                retry_counter = 0
+                LOGGER.debug("... list of tenants retrieved from auth.")
+            except (requests.exceptions.ConnectionError, requests.exceptions.Timeout) as connection_error:
+                LOGGER.warning(
+                    f"Connection error retrieving list of tenants: {connection_error}")
+                LOGGER.warning(f"Sleeping for {self.config.auth['timeout_sleep']} before trying again.")
+                time.sleep(self.config.auth["timeout_sleep"])
+                LOGGER.warning("Retrying...")
+                retry_counter -= 1
+            except requests.exceptions.TooManyRedirects as redirects_error:
```
Comment (raulnegreiros) [inline dojot/module/auth.py]: Is it not possible to catch a generic exception for others possible failures?

File: dojot/module/auth.py (hunk 2 above)
Comment (giovannicuriel) [reply dojot/module/auth.py]: It can be raise_for_status or just check error_code. I'll add it.

File: dojot/module/auth.py (hunk 3 above)
Comment (giovannicuriel) [reply dojot/module/auth.py]: It is not recommended to catch [a generic Exception](http://pylint-messages.wikidot.com/messages:w0703).

File: dojot/module/messenger.py (hunk 1 above)
Comment (giovannicuriel) [reply dojot/module/messenger.py]: Maybe not - if there is a problem with auth (not initialized, unreachable, etc.), messenger can't be initialized. Maybe the component might want to wait a little and then try to create a new messenger? 

As it is expected that messenger is able to know all tenants (or, at least, all tenants that the component is allowed to know), I think it's is a good thing to let the component know that it wasn't successfully initialized.

File: dojot/module/auth.py (hunk 3 above)
Comment (raulnegreiros) [reply dojot/module/auth.py]: Hum... ret.json raises an exception too, isn't?

**Pull request merged**